### DISCUSSION TITLE
Error Message: MongoError: bad auth Authentication failed through URI string

### DISCUSSION THREAD

**Discussion by arrullin** [community/community#36567] [Other Feature Feedback, Questions, & Ideas] [answered]
I'm trying to connect to my mongoDB server via the connection string given to me by mongo:

> "mongodb+srv://james:password@cluster0-re3gq.mongodb.net/test?retryWrites=true"

In my code I am calling the connection through mongoose like this (obviously putting in my password):
```javascript
const mongoose = require('mongoose');
const db = 'mongodb+srv://james:<mypassword>@cluster0-re3gq.mongodb.net/test?retryWrites=true'
mongoose
    .connect(db, { 
        useNewUrlParser: true,
        useCreateIndex: true
      })
    .then(() => console.log('MongoDB connected...'))
    .catch(err => console.log(err));
```

When I run the code I am getting the following error

```
MongoServerError: bad auth : Authentication failed.
    at Connection.onMessage (D:\Coding\Full Stack Website\node_modules\mongodb\lib\cmap\connection.js:207:30)
    at MessageStream.<anonymous> (D:\Coding\Full Stack Website\node_modules\mongodb\lib\cmap\connection.js:60:60)
    at MessageStream.emit (node:events:513:28)
    at processIncomingData (D:\Coding\Full Stack Website\node_modules\mongodb\lib\cmap\message_stream.js:132:20)
    at MessageStream._write (D:\Coding\Full Stack Website\node_modules\mongodb\lib\cmap\message_stream.js:33:9)
    at writeOrBuffer (node:internal/streams/writable:391:12)
    at _write (node:internal/streams/writable:332:10)
    at MessageStream.Writable.write (node:internal/streams/writable:336:10)
    at TLSSocket.ondata (node:internal/streams/readable:754:22)
    at TLSSocket.emit (node:events:513:28) {
  ok: 0,
  code: 8000,
  codeName: 'AtlasError',
  [Symbol(errorLabels)]: Set(1) { 'HandshakeError' }
}
```

Any ideas of what that could mean?

**Comment by santosharron** [comment, marked as answer]
You made the problem while connecting with MongoDB that is you made an error in this line
```javascript
const db = 'mongodb+srv://james:<mypassword>@cluster0-re3gq.mongodb.net/test?retryWrites=true'
```
It should be just the `password` not `<password>` so change that to
```javascript
const db = 'mongodb+srv://james:mypassword@cluster0-re3gq.mongodb.net/test?retryWrites=true'
```
It will work,

**Comment by diegoSilbestro** [comment]
I had the same error. The problem was that in mongoDb when i created the user whit his password i did not config his "buil-in role". You must to clic on edit and then in "buil-in role" select "read and write to any data base". 
I solved this problem whit this.

**Reply by ShubhamSShinde** [reply]
thank you for that help, I have been trying to fix this problem for 2 days

**Reply by levialkalidamat** [reply]
than'k, it is work fine for me

**Reply by aReynier** [reply]
Thank you! I was blocked on this problem since several hours

**Reply by GorHak33** [reply]
thank u man) you really help me)

**Reply by Muhammad-Ilyas-Ibrahim** [reply]
My username and password was correct and I tried for connection and connection was successful but I was not able to read and write. So I figured out that it's permissions issue.
And your comment saved my time.
Thanks buddy.

**Comment by debanjan-G** [comment]
ERROR:  MongoServerError: bad auth : authentication failed
 code: 8000,
codeName: 'AtlasError',
I am also encountering the same problem. But in my case I can connect to mongoDB atlas via shell. But whenever I am trying to deploy it via render, I am getting this error. Pls help.

**Reply by ashrth** [reply]
@debanjan-G Check my solution

**Comment by ashrth** [comment]
I resolved the error by changing the variable names, I had saved my database's username and password in .env file like this;
```
username=something
password=something
```
and I was connecting to the atlas using djongo, like this;
```
'default': {
        'ENGINE': 'djongo',
        'CLIENT': {
            'name': os.environ.get('DATABASE_NAME'),
            'host': os.environ.get('MONGO_URI'),
            'username': os.environ.get('DB_USERNAME'),
            'password': os.environ.get('DB_PASSWORD'),
            'authMechanism': 'SCRAM-SHA-1'

        }
```
so, when I changed by env variable to ```DB_USERNAME``` and ```DB_PASSWORD``` instead of just ```username``` and ```password```, the issue got resolved for me.

**Comment by ChinmayKaitade** [comment]
ERROR: MONGODB Connection Error MongoParseError: retryWrites must be either "true" or "false"

Please anyone give me solution for this...

**Comment by kyodgorbek** [comment]
I am also encounting same problem I was described here can anyone check this https://stackoverflow.com/questions/79570300/om-mongodb-mongocommandexception-command-failed-with-error-8000-atlaserror
